### PR TITLE
Add schema validation to base EFD client. 

### PR DESCRIPTION
**Pull request by kkeijuro**
Will throw a ValueError Exception when quering if topic doesnt exists in the schema

### CODE REVIEW

File: src/lsst_efd_client/efd_helper.py [hunk 1]
```diff
@@ -588,6 +597,25 @@ async def select_packed_time_series(
         vals.update({"times": df["times"]})
         return pd.DataFrame(vals, index=df.index)
 
+    async def _is_topic_valid(self, topic: str) -> bool:
+        """Check if the specified topic is in the schema.
+        A topic is valid and returns ``True`` if it is in the cached list of
+        topics. Any other case returns ``False``.
```
Comment (rra) [inline src/lsst_efd_client/efd_helper.py:603]: Single backticks can be used for Python core constants.
```suggestion
        A topic is valid and returns `True` if it is in the cached list of
        topics. Any other case returns `False`.
```

File: src/lsst_efd_client/efd_helper.py [hunk 2]
```diff
@@ -588,6 +597,25 @@ async def select_packed_time_series(
...
+        Parameters
+        ----------
+        topic : `str`
+            The name of the topic to look for.
+        Returns
+        -------
```
Comment (rra) [inline src/lsst_efd_client/efd_helper.py:609]: Needs blank lines before the headings.

**Review by rra — COMMENTED** [review]
We support dynamic topic addition, so I don't think caching the topics is safe. A long-running client may then throw spurious errors for topics that are indeed valid if they were added since the client started.

Other than that, just a couple of documentation nits; everything else looks good.

**Comment by kkeijuro** [comment]
> We support dynamic topic addition, so I don't think caching the topics is safe. A long-running client may then throw spurious errors for topics that are indeed valid if they were added since the client started.
> 
> Other than that, just a couple of documentation nits; everything else looks good.

If instead of doing this we check the existance or not of the topic by:

1.- do the normal query and get data
2.- check if data returned is empty
3a.- if its empty and doesnt exists then we raise the value exception
3b.- if it contains some values then we show the information

I guess then we will not have the sporious errors, what do you think? Btw Thanks for the PR

File: src/lsst_efd_client/efd_helper.py (hunk 2 above)
Comment (kkeijuro) [reply src/lsst_efd_client/efd_helper.py:609]: solved

File: src/lsst_efd_client/efd_helper.py (hunk 1 above)
Comment (kkeijuro) [reply src/lsst_efd_client/efd_helper.py:603]: solved

**Comment by rra** [comment]
Yes, that would work and would save on the request for topics in advance of sending the query in the normal case where the query returns some results. Thanks!

**Comment by kkeijuro** [comment]
Will you take a look at the PR?

Thanks

**Comment by rra** [comment]
This change is still caching the topics. I don't think it's safe to cache the topics at all, since we support dynamic topic creation.

**Comment by rra** [comment]
Each time there's an empty return and you have to decide whether to raise the exception, yes. It's unfortunate that it means another round trip to the server, but I think that's the only safe way to do it given that the list of topics is dynamic.

**Comment by kkeijuro** [comment]
Sorry, I've been busy this last days with another projects. I remove the topics cache.

Regards

**Comment by kkeijuro** [comment]
Hello,

Can you check the PR?

Thanks a lot

**Review by rra — APPROVED** [review]
This now looks good to me, thank you!

**Pull request merged**